djgeojson/views.py

```
@@ -40,6 +41,10 @@ def render_to_response(self, context, **response_kwargs):
         serializer = GeoJSONSerializer()
         response = self.response_class(**response_kwargs)
         queryset = self.get_queryset()
+
+        if type(queryset) is HttpResponseBadRequest:
+            return queryset
```

This is weird here, I believe we can raise a proper exception instead